lib/membrane_element_ffmpeg_h264/parser.ex

```diff
@@ -70,58 +112,39 @@ defmodule Membrane.Element.FFmpeg.H264.Parser do
   end
 
   @impl true
-  def handle_process(:input, %Buffer{payload: payload}, ctx, state) do
-    %{parser_ref: parser_ref, partial_frame: partial_frame} = state
-    payload = state.first_frame_prefix <> payload
+  def handle_process(:input, buffer, ctx, %{first_frame?: true} = state) do
+    buffer = Map.update!(buffer, :payload, &(state.first_frame_prefix <> &1))
+    handle_process(:input, buffer, ctx, %{state | first_frame?: false})
+  end
 
-    with {:ok, sizes} <- Native.parse(payload, parser_ref),
-         {bufs, rest} <- gen_bufs_by_sizes(partial_frame <> payload, sizes) do
-      state = %{state | partial_frame: rest}
-      actions = [buffer: {:output, bufs}, redemand: :output]
+  @impl true
```

Comment:
We've been using it for every clause to prevent cases where the last clause has a spelling mistake or different arity and is treated as a separate function without a warning

Reply:
BTW, I'm hoping that guy from Portland you've notified by mistake got used to notifications from Elixir projects :wink:

Reply:
Y'all are lucky I like Elixir! :upside_down_face: I actually discover a number of interesting projects, including this one, this way.

PS: I'm also waiting on https://github.com/membraneframework/membrane-protocol-icecast/pull/2 and https://github.com/membraneframework/membrane-element-rtp/pull/7 to land. :wink:

Reply:
And you discover the ones when somebody cares about using `@impl` ! you convinced me, I have just created https://github.com/moduledoc :) Speaking of the repositories, current plans are:
- icecast has quite low priority and probably won't be finished shortly, but we're glad to help if someone would contribute
- SRTP is one of our goals now, however we needed a quite big refactor of RTP to add RTCP first and they all will end up in https://github.com/membraneframework/membrane_rtp